src/julia_gc.c

```diff
@@ -513,7 +524,12 @@ static void TryMark(void * p)
 #endif
     }
     if (p2) {
-        JMark(p2);
+#ifdef DISABLE_BIGVAL_TRACKING
+        JMarkGapObjSafe(p2);
+#else
+        if (jl_typeis(p2, datatype_mptr))
+            JMark(p2);
```

Comment:
Why not always use this? Why is the `DISABLE_BIGVAL_TRACKING` doing something else?

Reply:
The two cases were flipped around, or alternatively, an `#ifndef` should have been used. So the `DISABLE_BIGVAL_TRACKING` was actually more restrictive.

Reply:
OK, but that doesn't really answer my question: why are the two cases doing something different? (Perhaps a code comment should explain it?)

Reply:
Okay, technically the `DISABLE_BIGVAL_TRACKING` should actually have a different name, namely that we only mark master pointers conservatively from the stack. Without the test, it would also mark small objects. I think the right thing to do here is to rename the define.